Print test case name in assert

rikardfalkeborn opened the pull request:
I sometimes find myself wondering which test cases I have broken when I fiddle with some check (since I then can try running these specific tests in the debugger, or make a small change and see if the tests pass). This PR adds the testclass and the test case names to the file and line number. I took special care to cover the case where an assert would be placed directly in the ```run()```-function, i.e., not inside a test case (from what I could see, no such case exists). If there is no need to handle this case, the code can be simplified (there wouldn't be need for the ```teardownTest()```-function for example).

The exact format for how to print the test name is very much up for discussion.

danmar commented:
> I took special care to cover the case where an assert would be placed directly in the run()-function,

ok. I vote to remove this to simplify the code.

rikardfalkeborn commented:
> > I took special care to cover the case where an assert would be placed directly in the ruby lastn()-function,
> 
> ok. I vote to remove this to simplify the code.

Removed. I also changed the way the test name was printed slighly to make it easier to distinguish from file and line.

danmar commented:
I merged this with 204ce795b

danmar commented:
Thanks!
